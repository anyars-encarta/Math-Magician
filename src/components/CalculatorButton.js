import React from 'react';
import PropTypes from 'prop-types';

function CalculatorButton({ children, onClick, className }) {
  return (
    <button type="button" className={className} onClick={onClick}>
      {children}
    </button>
  );
}

CalculatorButton.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
};

CalculatorButton.defaultProps = {
  className: '',
};

export default CalculatorButton;
