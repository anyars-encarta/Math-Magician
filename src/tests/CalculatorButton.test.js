import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import CalculatorButton from '../components/CalculatorButton';

describe('CalculatorButton', () => {
  test('renders correctly', () => {
    const { container } = render(
      <CalculatorButton onClick={() => {}}>
        Test Button
      </CalculatorButton>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('calls onClick callback when clicked', () => {
    const onClickMock = jest.fn();
    const { getByText } = render(
      <CalculatorButton onClick={onClickMock}>
        Test Button
      </CalculatorButton>,
    );
    fireEvent.click(getByText('Test Button'));
    expect(onClickMock).toHaveBeenCalled();
  });
});
