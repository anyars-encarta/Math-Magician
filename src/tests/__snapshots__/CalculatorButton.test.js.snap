// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorButton renders correctly 1`] = `
<button
  class=""
  type="button"
>
  Test Button
</button>
`;
